src/lua/utils.c

```diff
@@ -54,6 +54,43 @@ uint32_t CTID_UUID;
 uint32_t CTID_DATETIME = 0;
 uint32_t CTID_INTERVAL = 0;
 
+/** A copy of index2adr() from luajit/src/lj_api.c. */
+static TValue *
+index2adr(lua_State *L, int idx)
+{
+	if (idx > 0) {
+		TValue *o = L->base + (idx - 1);
+		return o < L->top ? o : niltv(L);
+	} else if (idx > LUA_REGISTRYINDEX) {
+		api_check(L, idx != 0 && -idx <= L->top - L->base);
+		return L->top + idx;
+	} else if (idx == LUA_GLOBALSINDEX) {
+		TValue *o = &G(L)->tmptv;
+		settabV(L, o, tabref(L->env));
+		return o;
+	} else if (idx == LUA_REGISTRYINDEX) {
+		return registry(L);
+	} else {
+		GCfunc *fn = curr_func(L);
+		api_check(L, fn->c.gct == ~LJ_TFUNC && !isluafunc(fn));
+		if (idx == LUA_ENVIRONINDEX) {
+			TValue *o = &G(L)->tmptv;
+			settabV(L, o, tabref(fn->c.env));
+			return o;
+		} else {
+			idx = LUA_GLOBALSINDEX - idx;
+			return idx <= fn->c.nupvalues ?
+			       &fn->c.upvalue[idx - 1] : niltv(L);
+		}
+	}
+}
+
+GCcdata *
+get_cdata(struct lua_State *L, int idx)
```

Comment:
OK, this is not quite the thing I meant in the previous review. I propose to implement a new function `luaL_tocpointer(lua_State *L, int idx, uint32_t *ctypeid)` with the signature similar to `lua_tolstring` and implementing `luaL_checkcdata` without actual check. I left an example below.

```c
void *
luaL_tocpointer(lua_State *L, int idx, uint32 *ctypeid) {
        TValue *o = index2adr(L, idx);
        if (!tviscdata(o)) {
                if (ctypeid != NULL) *ctypeid = 0;
                return NULL;
        }
        GCcdata *cd = cdataV(o);
        if (ctypeid != NULL) *ctypeid = cd->ctypeid;
        return cdataptr(cd);
}
```

IMHO, such function can be used widely in Tarantool to work with cdata without additional checks.

Reply:
Done.